src/v/config/base_property.h

```diff
@@ -75,7 +79,8 @@ class base_property {
     // this serializes the property value. a full configuration serialization is
     // performed in config_store::to_json where the json object key is taken
     // from the property name.
-    virtual void to_json(json::Writer<json::StringBuffer>& w) const = 0;
+    virtual void to_json(
+      json::Writer<json::StringBuffer>& w, redact_secrets redact) const = 0;
```

Comment:
it looks like there might be some other users of the interface that are updated in the first commit

```
../../../src/v/config/tests/config_store_test.cc:231:27: error: too few arguments to function call, expected at least 2, have 1
    cfg.to_json(cfg_writer);
    ~~~~~~~~~~~           ^
```

Reply:
Ah nice catch. I broke the original commit up by file and didn't come back to re-add test changes.